### ISSUE TITLE
s.js not compatible with IE 11 due to String.endsWith usage

### ISSUE THREAD

**Issue by rdmurphy** [systemjs/systemjs#1962]
Hello!

So if there is no intention for `s.js` to be usable in IE 11, then ignore this! But I _think_ the usage of `String.endsWith` is breaking IE support, because that doesn't exist in that browser.

The error I am seeing (I have a Promise polyfill in place):

```
Possible Unhandled Promise Rejection: TypeError: Object doesn't support property or method 'endsWith'
```

If this is a real issue, it may be good to either 1) remove that call, or 2) document we need to polyfill that too.

**Comment by rdmurphy** [comment]
Can confirm that once I added the [`endsWith` polyfill on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith#Polyfill) the s.js loader began working again.

**Comment by guybedford** [comment]
Yes IE11 is supposed to be supported, this is a regression in 4.0.0.

The fix should be just using `substr` or similar in the JSON detection line.

Apologies for the issue - will aim to get a patch out next week.

**Comment by joeldenning** [comment]
Fixed in 4.0.1. See https://github.com/systemjs/systemjs/blob/master/CHANGELOG.md
